src/features/BookSeats/ui/BookSeatsBlock.tsx

```diff
@@ -1,33 +1,102 @@
-import { FC } from 'react';
+import { FC, useEffect, useState } from 'react';
 
 import { useParams } from 'react-router-dom';
 
+import { useCreateReservationMutation } from 'entities/reservation/hooks/useCreateReservation';
 import { useGetSession } from 'entities/session/hooks/useGetSession';
 import { ErrorWrapper, LoadingSpinner } from 'shared/ui';
 
+import { Seat } from '../types/seat';
+import { CountdownTimer } from './CountdownTimer';
 import { HallDesign } from './HallDesign';
 import { OrderInformation } from './OrderInformation';
+import { ReservationExpiredDialog } from './ReservationExpiredDialog';
 
 export const BookSeatsBlock: FC = () => {
-    const { id } = useParams();
+    const { id: sessionId } = useParams();
+    const { data: session, isError, isLoading } = useGetSession(sessionId!);
+    const [chosenSeats, setChosenSeats] = useState<Seat[]>([]);
+    const [reservationExpirationDate, setReservationExpirationDate] = useState<Date | null>(null);
+    const [reservationId, setReservationId] = useState('');
+    const [secondsRemaining, setSecondsRemaining] = useState<number | null>(null);
 
-    const { data: session, isError, isLoading } = useGetSession(id!);
+    const onSeatClick = (seat: Seat) => {
+        if (chosenSeats.includes(seat)) {
+            setChosenSeats((prev) => prev.filter((p) => p !== seat));
+        } else {
+            setChosenSeats((prev) => [...prev, seat]);
+        }
+    };
+
+    const { mutate: createReservation } = useCreateReservationMutation();
+
+    useEffect(() => {
+        if (session) {
+            createReservation(
+                { total_price: 0, sessionId: session.id },
+                {
+                    onSuccess: (reservation) => {
+                        const expirationDate = new Date(reservation.expires_at);
+                        setReservationExpirationDate(expirationDate);
+                        setReservationId(reservation.id);
+                    }
+                }
+            );
+        }
+    }, [session]);
+
+    useEffect(() => {
```

Comment:
It looks like this useEffect + const [secondsRemaining, setSecondsRemaining] = useState<number | null>(null); can be moved to the shared hook with name like useTimer or something and make it generic